types/mydecimal.go

```diff
@@ -432,6 +432,9 @@ func (d *MyDecimal) FromString(str []byte) error {
 	wordsInt, wordsFrac, err := fixWordCntError(wordsInt, wordsFrac)
 	if err != nil {
 		digitsFrac = wordsFrac * digitsPerWord
+		if err == ErrTruncated {
```

Comment:
The truncated error should not be ignored here. There is an example:

```sql
err = sc.HandleTruncate(res.FromString([]byte(val)))
```

we'd better handle it out of this function.